src/main/java/com/ledikom/service/BotService.java

```diff
@@ -66,6 +67,19 @@ public void processCouponsInMap() {
         CouponService.userCoupons.entrySet().removeIf(userCoupon -> userCoupon.getValue().getExpiryTimestamp() < System.currentTimeMillis() - 5000);
     }
 
+    @Scheduled(cron = "0 0 0 * * *")
```

Comment:
Try smth like 
`@Scheduled(cron = "0 0 21 * * *", zone = "Europe/Moscow")`
scheduled annotation uses UTC time, not server time.
So it is going to be 00:00 or 01:00 depending on season, which is ok in our case